### PR TITLE
Solution Restore should run in background

### PR DESCRIPTION
**Pull request by alpaix**
Resolves NuGet/Home#3816.
Resolves NuGet/Home#4255.
Resolves NuGet/Home#4636.

In order to prevent hangs in VS Solution Restore should create a background task and push it in the restore worker queue.

Added E2E test suite for .NET Standard multi-targeting class library

//cc @rrelyea

### CODE REVIEW

**Review by emgarten — APPROVED** [review]
🚀

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs [hunk 1]
```diff
@@ -103,25 +105,23 @@ internal sealed class SolutionRestoreCommand
...
+            if (_restoreTask.IsCompleted)
             {
-                // QueryStatus should disable the context menu in most of the cases.
-                // Except when NuGetPackage was not loaded before VS won't send QueryStatus.
-                Logger.Log(MessageLevel.Info, Resources.SolutionRestoreFailed_RestoreWorkerIsBusy);
+                _restoreTask = NuGetUIThreadHelper.JoinableTaskFactory
+                    .RunAsync(() => SolutionRestoreWorker.ScheduleRestoreAsync(
+                        SolutionRestoreRequest.ByMenu(),
+                        CancellationToken.None))
+                    .Task;
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs:114]: Would it be able to cancel this task when close solution or close VS is performed during solution restore? or could it make it hang?

Comment (alpaix) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs:114]: The worker itself will take care of cancelling the task. Tested this particular scenario manually by closing the solution while restore is in progress.

**Review by jainaashish — APPROVED** [review]
Looks good

**Comment by rrelyea** [comment]
@alpaix - the 3 issues that this PR resolve are all in Future-0, and now closed. Can you tell me which release this commit shipped in, please.

**Pull request merged**